### PR TITLE
fix(DB/Zone):improved Eastern Plaguelands

### PR DESCRIPTION
**Pull request by Knindzagxg**
Eastern Plaguelands are now fully updated, since some creatures in zone was missing SmartAI.

Working on every zone at the moment, and will post updates once i'm done with each.

To test it:
Apply SQL
Restart your server

You can manually spawn NPC by ID to check SmartAI or you can go to zone and check like that as well.

.npc add ID
or
.tele to zone and run / test around.

### CODE REVIEW

**Review by Si1ker — APPROVED** [review]
Looks good, I'll have this tested today at night. 🐱

**Comment by Si1ker** [comment]
Only one failed, included is the change required in order to make it work. 🐱 



Scourge Siege Engineer
image


Scarlet Inquisitor = Wrong   @Knindzagxg 
They never use it because they only have health.
image

I have found why.. we're gonna need this change since the spell costs 300p mana, I did the change and it goes in accordance with his new SAI
```sql
UPDATE `creature_template` SET `unit_class`=2 WHERE  `entry`=15162;
```
Do you want to include the change here o do you want me to do a quick PR for it? ping me on dc for this.

Living Decay - Between 40-80% Health
image


Mossflayer Zombie - In Combat
image


Plaguehound Runt - In Combat
image


Large Vile Slime - On Aggro
image

Large Vile Slime - Between 30-60% Health
image

Large Vile Slime - Between 10-20% Health
image

**Comment by Kitzunu** [comment]
The question is...
A) should he cast it
B) should he have mana?
C) i believe there is a flag to ignore spell cost 👍

**Comment by Knindzagxg** [comment]
Awesome tests mate.

We can include / change, tho think Kiz is right.
There should be flag for casting, but will check once im home.

**Comment by Knindzagxg** [comment]
Updated unit_class.

Wasn't able to find mask ID to ignore mana requirements.
Also, trough wowhead comments people confirming they will cast it.

File: data/sql/updates/pending_db_world/rev_1601314356612298800.sql [hunk 1]
```diff
@@ -12,6 +12,7 @@ INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_
 (17878, 0, 0, 0, 4, 0, 100, 0, 0, 0, 0, 0, 11, 7978, 0, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Scourge Siege Engineer - On Aggro - Cast \'7978\'');
 
 UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 15162;
+UPDATE `creature_template` SET `unit_class`= '2' WHERE `entry`=15162;
```
Comment (Helias) [inline data/sql/updates/pending_db_world/rev_1601314356612298800.sql:15]: don't use strings when integers are expected

**Review by Helias — CHANGES_REQUESTED** [review]

**Comment by Si1ker** [comment]
@Knindzagxg  @Kitzunu  Just to keep records that other options were tried.

Tried unit_flag

32 | 0x00000020 | UNIT_FLAG_PREPARATION | Don't take reagents for spells with SPELL_ATTR_EX5_NO_REAGENT_WHILE_PREP
-- | -- | -- | --



Tried unit_flags2

262144 | 0x00040000 | UNIT_FLAG2_ALLOW_CHEAT_SPELLS | allows casting spells with AttributesEx7 & SPELL_ATTR7_IS_CHEAT_SPELL
-- | -- | -- | --


Only the change suggested (which Knindzagxg already changed here) will work with the new SAI.

@Helias  String numeric value fixed, PR good to go !

**Review by Si1ker — APPROVED** [review]

**Comment by Knindzagxg** [comment]
Yeah, tried few my self but no luck sadly.

**Review by Helias — APPROVED** [review]

**Pull request merged**